BlazarService/src/main/java/com/hubspot/blazar/listener/MetricBuildVisitor.java

```diff
@@ -0,0 +1,40 @@
+package com.hubspot.blazar.listener;
+
+import java.util.SortedMap;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.hubspot.blazar.base.ModuleBuild;
+import com.hubspot.blazar.base.RepositoryBuild;
+import com.hubspot.blazar.base.visitor.ModuleBuildVisitor;
+import com.hubspot.blazar.base.visitor.RepositoryBuildVisitor;
+
+@Singleton
+public class MetricBuildVisitor implements ModuleBuildVisitor, RepositoryBuildVisitor {
+
+  private final MetricRegistry metricRegistry;
+
+  @Inject
+  public MetricBuildVisitor(MetricRegistry metricRegistry){
+    this.metricRegistry = metricRegistry;
+    for (ModuleBuild.State state : ModuleBuild.State.values()) {
+      metricRegistry.register(ModuleBuild.class.getName() + "." + state.toString().toLowerCase(), new Meter());
+    }
+
+    for (RepositoryBuild.State state : RepositoryBuild.State.values()) {
+      metricRegistry.register(RepositoryBuild.class.getName() + "." + state.toString().toLowerCase(), new Meter());
+    }
+  }
+
+  public void visit(ModuleBuild moduleBuild) {
+    SortedMap<String, Meter> meters = metricRegistry.getMeters();
+    meters.get(moduleBuild.getClass().getName() + "." + moduleBuild.getState().toString().toLowerCase()).mark();
```

Comment:
I prefer `name()` over `toString()` for this in case someone overrides `toString()`. Also you don't need to pre-register the meters, you can just do something like `metricRegistry.meter(metricName).mark()` inside this method and remove the pre-registration form the constructor